zboxcore/sdk/sdk.go

```diff
@@ -1119,7 +1120,9 @@ func smartContractTxnValueFee(sn transaction.SmartContractTxnData,
 	txn.TransactionFee = fee
 	txn.TransactionType = transaction.TxnTypeSmartContract
 
-	if err = txn.ComputeHashAndSign(client.Sign); err != nil {
+	if err = txn.ComputeHashAndSign(func(hash string) (string, error) {
+		return sys.Sign(hash, client.GetClient().SignatureScheme, client.GetClientSysKeys())
```

Comment:
I saw several places that use this sign callback function, we can move this function to client.Sign, and use client.Sign here just like what it did before to make this cleaner.

Reply:
reverted